### PR TITLE
RTL alignment

### PR DESCRIPTION
**Pull request by jpp-odoo**
[FIX] web: invert position (start/end) when using rtl

Before this commit, when using a rtl language, the position of the
popper elements using the 'usePosition' hook weren't inverted.

---

[FIX] web: align text to the end in RTL languages

Bootstrap forces some input types ("tel", "url", "email", "number") to
be LTR even if in an RTL languages, before the last update of bootstrap
this was done in [1] and [2].

The issue of forcing some input types to LTR, is that the alignment was
wrong. Now, the for the same types when using an RTL language we force
the align to the end.

[1] : https://github.com/odoo/odoo/commit/90ab752c5fc9b27a6a40006abfe49483e7ca847d
[2] : https://github.com/odoo/odoo/commit/0d650c3b19c847181fa50c88eb945792528f01cc

### CODE REVIEW

**Review by aab-odoo — APPROVED** [review]
@brboi maybe?

File: addons/web/static/src/core/position_hook.js [hunk 1]
```diff
@@ -81,7 +82,14 @@ const DEFAULTS = {
...
     // Retrieve directions and variants
-    const [directionKey, variantKey = "middle"] = position.split("-");
+    let [directionKey, variantKey = "middle"] = position.split("-");
+    if (localization.direction === "rtl") {
+        if (variantKey === "start") {
+            variantKey = "end";
+        } else if (variantKey === "end") {
+            variantKey = "start";
+        }
+    }
```
Comment (brboi) [inline addons/web/static/src/core/position_hook.js:92]: Do we really want to switch between right-end and right-start and also between left-start and left-end when we are in RTL ?
I don't think so.

IMHO, we should not make the position hook dependent of the localization.

The strategy used in the Dropdown component is to pass the proper positioning options depending on the l10n direction. I think this is what we should do when we face a similar use case.

File: addons/web/static/tests/core/position_hook_tests.js [hunk 2]
```diff
@@ -176,6 +177,24 @@ QUnit.test("position right", getPositionTest("right"));
...
+        direction: "rtl",
+    });
+    await getPositionTest("top-middle")(assert);
+});
+QUnit.test("position RTL top-end", async function (assert) {
+    patchWithCleanup(localization, {
+        direction: "rtl",
+    });
+    await getPositionTest("top-end", "top-start")(assert);
+});
```
Comment (brboi) [inline addons/web/static/tests/core/position_hook_tests.js:197]: As this test suite is already exhaustive, I would like to see it evolve in the same way.
Add tests for:
- bottom-{start,middle,end}
- right-{start,middle,end}
- left-{start,middle,end}

**Review by brboi — CHANGES_REQUESTED** [review]

File: addons/web/static/src/core/dropdown/dropdown.js [hunk 3]
```diff
@@ -95,17 +94,9 @@ export class Dropdown extends Component {
...
         /** @type {string} **/
-        let position =
+        const position =
             this.props.position || (this.parentDropdown ? "right-start" : "bottom-start");
-        let [direction, variant = "middle"] = position.split("-");
-        if (localization.direction === "rtl") {
-            if (["bottom", "top"].includes(direction)) {
-                variant = variant === "start" ? "end" : "start";
-            } else {
-                direction = direction === "left" ? "right" : "left";
```
Comment (brboi) [inline addons/web/static/src/core/dropdown/dropdown.js:105]: This line is still needed for the direction caret class, am I right ?
Or maybe the directionCaretClass have to be set on the "onPositioned" callback.

**Review by brboi — COMMENTED** [review]
It already looks nice, thanks ;-)
Please investigate the below comment.
I'll give a try at an instance after.

**Comment by brboi** [comment]
robodoo r+

**Comment by jpp-odoo** [comment]
@robodoo rebase-ff

**Pull request closed without merging**